### ISSUE TITLE
[Please Comment] Next Sample Module for Serenity

### ISSUE THREAD

**Issue by volkanceylan** [volkanceylan/Serenity#715]
As TypeScript conversion is complete (almost, excluding Serenity.Script.UI), i'm now thinking about what to do next with Serenity as a sample module, like Northwind.

I have some ideas, like some i'll list below, but want to hear your opinions. Maybe somebody will come back with a better option that will be more useful.

If you'll offer some other project type, please include links to a sample app and its database schema.

- *Adventure Works*: not sure about this one, feels like just a bit more complex Northwind 

- *Meeting Management*: this is something i've done before for corporate meetings. agenda, decisions, e-mail invitations, pdf meeting report etc.

- *Task / Issue / HelpDesk System*: another one i have experience with. my solution had multi level tree tasks like ToDoList (windows app from codeproject). workflow etc. maybe i'll keep it simple and model it similar to GitHub or GitLab.

- *E-Commerce System*: Don't have a schema at the moment, but there are so many to take as sample

- *CMS*: Content Management Systems anyone?

- *CRM*: Sample from Microsoft Dynamics CRM, Sugar CRM, ...?

- *Dynamic Entities*: Just like SharePoint or Dynamics CRM, design your tables, fields, forms in app, 
and have a dynamic interface

- *Database Browser*: something like phpmyadmin (if i remember name right)

- *Forum / Blog*: seems too easy
- Student / Academic Information Management System: something i do in our university

Ok, what do you think?

**Comment by rolembergfilho** [comment]
Hey,

What about a physical therapy program?

**Comment by deivisonrpg** [comment]
HelpDesk System
CRM
Dynamic Entities
Is great +1

**Comment by fazanan** [comment]
Dynamic Enitities, might be complex but we can more learn from this.

**Comment by awesomegithubusername** [comment]
Hi Volkan;

I'd say go for "Task / Issue / HelpDesk System". I've been searching for a while for some good ASP.NET implementation, to little avail. I can help in the "Microsoft Dynamics CRM" department, though, as I'm currently designing an MS CRM (version 4, but it can easily be ported to other versions) Web Portal for our clients. How about implementing a messaging app (admin LTE)?

**Comment by volkanceylan** [comment]
Another question, should it be implemented as tutorial or embedded?

**Comment by awesomegithubusername** [comment]
It depends on the chosen sample(HelpDesk|Meeting Management---->tutorial, messaging|Adventure Works----->embedded).

**Comment by jsbUSMC** [comment]
I think the HelpDesk / Task / Issue system would be a really good addition. There aren't many systems that are OSS that implement a good drop-in task management application. It could be a big boost to our users, and I know I would love it. We could implement SignalR for real-time updates and notifications.

**Comment by jsbUSMC** [comment]
I also think that the Helpdesk / Meeting management should be implemented as tutorial.

**Comment by PanChenhao** [comment]
I vote for HelpDesk System.

**Comment by samdubey** [comment]
I vote for Meeting Management System

**Comment by samdubey** [comment]
Then CRM

**Comment by marcobisio** [comment]
HelpDesk System would be great!
I also pick the "tutorial way"...
Thanks!

**Comment by ramveersgh** [comment]
For me too HelpDesk and Meeting Management seems to be good for sample.
HelpDesk may be the embedded with in template. 
Tutorial for ISSUE TRACKER may be also a good option.

There may be one sample with editing on page itself. 
For Example : Employee Profile which contains some fields like Name, address, email, contacts etc and some grids like Experience, Qualification, Skills etc..
Currently samples are only for ChangePassword and SignUp page which is not enough to build complex pages.

**Comment by volkanceylan** [comment]
Can you sample which kind of HelpDesk you imagine. Sample application or schema

**Comment by awesomegithubusername** [comment]
A simple help desk(Issue Tracker) a la GitHub (Tickets, Assigned to, Due, Status(open|resolved), Priority(Normal|Critical|...), Category(Technical|Sales|..., Knowledge Base(wiki...)

https://www.jitbit.com/hosted-helpdesk/

**Comment by Estrusco** [comment]
Hi! 

I vote for 
•Meeting Management: 
•Task / Issue / HelpDesk System
•CRM

For me tutorial is great.

Thank you!

**Comment by wdhodges** [comment]
My Vote:

* Help Desk / Issue Tracker / Task system
* I would really love to see dynamic entities
* Meeting management

I think those 3 things could drive the current community to new heights. Also I feel like if you keep building the system like you are building it you might (and should!) get picked by my Microsoft. This truly is a Lightswitch alternative framework.

**Comment by chunlizh** [comment]
Hi Volkan,
I vote for :

- E-Commerce System
- CMS
- CRM
- Dynamic Entities

Thanks!

**Comment by tvldev02** [comment]
Task / Issue / HelpDesk System

**Comment by steax85** [comment]
hi Volkan,

I vote for "Task / Issue / HelpDesk System"

thank you!

**Comment by volkanceylan** [comment]
Thanks to all that commented so far. After evaluating your suggestions, i'm now planning to start with meeting as a tutorial followed by Task management as an integrated module.

**Comment by marcobisio** [comment]
A bit off topic, but it would be great also a sample/how-to for the user preferences added in the 2.1.5 release...

Thanks!

**Comment by rolembergfilho** [comment]
I think that it will be more powerfull and interesting if existing Serenity developer share their implementation in other application features, like custom error validator, dashboard, etc...

**Comment by volkanceylan** [comment]
@marcobisio there is already a topic for that:

https://volkanceylan.gitbooks.io/serenity-guide/content/grids/persisting_settings.html

**Comment by marcobisio** [comment]
@volkanceylan I've missed that...thank you very much!!!

**Comment by samdubey** [comment]
Eagerly waiting to see new tutorial

**Comment by RouteTech** [comment]
how about a wizard generator

**Comment by qfleurat** [comment]
CMS:
Dynamic Entities:

**Comment by AtillaBilgic** [comment]
Hi @volkanceylan 

According to me either 
- Task / Issue / HelpDesk System
or
- Meeting Management

as Tutorial will be great.
Thanks

**Comment by wdhodges** [comment]
When is the estimated time for these to be released? Just wondering :)

**Comment by samdubey** [comment]
Please see updates at Serenity Guide and https://github.com/volkanceylan/MeetingManagement.git
Development kicked off

**Comment by alexsandrocruz** [comment]
Hi mr. @volkanceylan,

my vote goes to:

CRM,
Lead Management System (like Mautic),
CMS

So long and thanks for all the fish

**Comment by BravoSierra** [comment]
Hi, Volkan.
I've just discovered Serenity recently, thanks for this superb software.
As for your request for suggestions, it might be interesting to start from an existing Open Source solution (e-commerce or CMS) and do things diffently and/or add some functionnalities or UI
that it does not have naturally. So you might have an immediate interest from these communities and get Serenity to be know more quickly (my personnal example, I've heard of Serenity throught the Lightswitch blogs and I've already posted there that everybody should give it a try.

For example: 

Take nopCommerce.com e-commerce and manage the backoffice with Serenity.
Start from DNN CMS (dnnsoftware) and make a backoffice with Serenity
(there are probably better ideas but you get the point that is starting with an existing project/community and extend it in a usefull way to get people's interest.

**Comment by volkanceylan** [comment]
@wdhodges i could only start Meeting tutorial than had to look at adding more samples to Serene. will try to return back to guide soon.

@BravoSierra, thanks for the idea. probably i should look for a package that i had experience with for integration sample, but i got the point. I have plans to add Lightswitch/Dynamics CRM like, UI based app development ability. I didn't try Lightswitch actually, but i understand that it has similarities to Serenity. starting with targeted project types, e.g. LOB.

**Comment by BravoSierra** [comment]
The short story is that LightSwith started as a Microsoft projet using Silverlight and with a marketing position like Access, ie for the non-technical people wanting an easy small database  tool. 

Then with SilverLight beeing deprecated sooner or later, MS LS team made a 'mobile' Lightswitch and said that the great new thing about LS was the possibility to create a unique project for both targets and build it either as a Silverlight LOB deskop app or as a mobile App. 

Then LS abandonned LightSwitch without an explanation and left the community orphan. For this reason I believe that many Lightswitchers are or will be willing to find a replacement for LS and Serenity could be a very good candidate to welcome the migrants.

The discussion about Serenity is here ; you might want to have a look from time to time since there is regularly a new discussion in the forums about the future of LS

https://social.msdn.microsoft.com/Forums/vstudio/en-US/aa613e75-b51d-40b1-9eb6-7a929f1ff9be/serenity?forum=lightswitch

**Comment by volkanceylan** [comment]
I wonder if it would be possible to automatically convert Lightswitch app to Serenity one :) 

Unfortunately MS has a bad track of deprecating tools with potential like Silverlight and LightSwitch. But somehow OLE, COM, XML still lives at the core of Visual Studio.

**Comment by xjbsky** [comment]
I like Task / Issue / HelpDesk System and hope it  implemented as tutorial

**Comment by xjbsky** [comment]
I like serenity, though i only learn it at this year, but it is excellent, and i use it as a framework to begin my serveral app, i hope it be simple and focused on as a framework

**Comment by SimeonDD** [comment]
I vote for **none** Volkan.

I have extensive experience with non-website coding and some with Silverlight.
Your product allows me to enter the web app field very rapidly. You shrink the development from several days for a single page to probably half a day. Your code is clean.

You also offer a combination that is **extremely** rare:
1.	Your demo project compiles and runs from the get-go.
2.	The visual appeal is very nice and workable without any additional work.
3.	The documentation is very nice.
4.	The current template is absolutely enough for anyone to port into their own business areas.

Proof that Serene is great? You have created a following.

And none of this matters if Serenity disappears because of lack of funding, focus and inability to scale for the demand.

I think you should put your attention in getting a paid subscription off the ground as soon as possible.
This will give corporate users security and will provide funding for development, tutorials and online support.

The corporate users will not mind paying and will prefer to have some long-term security this way. All that will add more value for the free subscribers.

I am evaluating Serene for an employer. I absolutely love it but at the same time I am very uneasy because it all feels like a one-man show. 

It seems to me that Serene is now out of the “cute” phase. At this point you have to make sure this thing does not die on us.

What will Serenity be?
A) An industry standard that:
-	Reduces cost of development.
-	Offers common platform for faster knowledge transfer and long-term support.
-	Decreases the time to market.
-	Outsources the specific business field knowledge as a responsibility of the developer.

B) A selection of half-backed starter templates for different business needs.

C) A tap-on-the-back and feel-good forum.

Basically – OK, you’ve got people’s attention. Now what?

**Comment by BravoSierra** [comment]
Hello Simeon. I agree with most of what is said above but one should be aware that switching to a subscription model is not an easy task and must be done very carefull. 

This is what Shaun Walker says in his book about the attempt to set up a subscription model for DotNetNuke in 2003(!). The chapter is "Subscription fiasco". Exerpt: _"... I had to face the wrath of a scorned community. Among other things, I was accused of misleading the community, lying about the open source nature of the project, and leting my personal greed cloud my vision. For one supporter of my decision, there were 10 more who publicly crucified me as the evil incarnate_", etc
 (Professional DNN7, Open Source .NET CMS Platform, Wrox editor)

Shaun made a 180 degrees backtrack and only much later, DNN developped a model with a pro and a community version, but this is a difficult strategy because you have to maintain 2 versions.
 I did not like this move either.

So while a understand your concern that a project must ensure it's Financial viabilité for the future, I suggest other stratégies to be considered : paid skins or modules; paid full technical documentation (this is what nopcommerce.com does for about 80 euros you get the PDF); paid consulting, paid partnership program (you are listed for your country as a 'serene' specialist, etc

I would let the community grow up and not take the risk to break the momentum too early.

Apart from this I'm 100% with you on serenity qualities and potential.

To me what it needs most now is a forum apart from the issue tracker.

Thanks Volkan and continue the good work! (while taking some vacation if you can?)

**Comment by SimeonDD** [comment]
GREAT!
I used "subscription based" too loosely! Please consider the meaning to be "Something of your choice that would support the life of that wonderful product". 

About the label "greed" too easily dispensed from the comfort of ones home. 

Normal thinking reveals a simple fact. Things happen in time and space. 
Fact-1: You do not have 24h/d available for work. 
Fact-2: You have to live and pay bills. 
Fact-3: You have to divide the rest between self-education, managing, coding and manning this forum (it takes on average 20 min of **_total_** time to write half a page coherent response.). 
Fact-4: The way a starving athlete cannot compete, the same way a stressed out brain suffers shrinking bandwidth and produces bad decisions.

So, disregard opinions that disregard these facts because if you disregard these facts Serenity is dead and that serves nobody.

Paid consulting is a fine alternative.
Maintaining two versions is really not that bad if you have the cash to support it.
Profit sharing with creators of skins is fine too.
Paid docs ... hm ... PDF not good - here is why:
- Too easy to copy for free. 
- You want growth and that comes with availability of documentation.
- Kind of nasty but still true ... perfect documentation suffocates paid consulting.
- Even if online-only and restricted, tutorial content **_will be_** copied and used for free.

OK ... do whatever you want, Just try to keep this thing alive man.

**Comment by RMamoon** [comment]
@volkanceylan Thanks for this superb template. You really done very good job for most of people who are looking for initial template to implement their own business needs. And I am one of them. I really appreciate your efforts. After that I will suggest to add modules that will fulfill needs for manufacturing industries business functionalities.  including MRP , Sales, Purchases, Chart of accounts, Financial reporting etc It could be very helpful for many of developers that normally look into open source solutions. I would like to participate in these modules.

**Comment by volkanceylan** [comment]
Thanks for all the ideas. I took a note of them, and will try to implement some, or make use partially. I certainly want to have some complete apps to demonstrate real world usage of Serenity. Some content / support / options might become premium, didn't decide yet, but Serenity will always have a free, open, community edition.
